searx/engines/google_images.py

```diff
@@ -153,76 +95,18 @@ def response(resp):
-    #     In this structure the link to the origin PNG, JPG or whatever is given
-    # first link per image-div contains a <img> with the data-iid for bas64 encoded image data::
-    #      <img class="rg_i Q4LuWd" data-iid="0"
-    # second link per image-div is the target link::
-    #      <a class="VFACy kGQAp" href="https://en.wikipedia.org/wiki/The_Sacrament_of_the_Last_Supper">
-    # the second link also contains two div tags with the *description* and *publisher*::
-    #      <div class="WGvvNb">The Sacrament of the Last Supper ...</div>
-    #      <div class="fxgdke">en.wikipedia.org</div>
-
-    root = eval_xpath(dom, '//div[@id="islmp"]')
-    if not root:
-        logger.error("did not find root element id='islmp'")
-        return results
-
-    root = root[0]
-    for img_node in eval_xpath_list(root, './/img[contains(@class, "rg_i")]'):
-
-        img_alt = eval_xpath_getindex(img_node, '@alt', 0)
-
-        img_base64_id = eval_xpath(img_node, '@data-iid')
-        if img_base64_id:
-            img_base64_id = img_base64_id[0]
-            thumbnail_src = img_bas64_map[img_base64_id]
-        else:
-            thumbnail_src = eval_xpath(img_node, '@src')
-            if not thumbnail_src:
-                thumbnail_src = eval_xpath(img_node, '@data-src')
-            if thumbnail_src:
-                thumbnail_src = thumbnail_src[0]
-            else:
-                thumbnail_src = ''
-
-        link_node = eval_xpath_getindex(img_node, '../../../a[2]', 0)
-        url = eval_xpath_getindex(link_node, '@href', 0, None)
-        if url is None:
-            logger.error("missing @href in node: %s", html.tostring(link_node))
-            continue
-
-        pub_nodes = eval_xpath(link_node, './div/div')
-        pub_descr = img_alt
-        pub_source = ''
-        if pub_nodes:
-            pub_descr = extract_text(pub_nodes[0])
-            pub_source = extract_text(pub_nodes[1])
-
-        src_url = get_img_url_by_data_id(data_id_to_img_url, img_node)
-        if not src_url:
-            src_url = thumbnail_src
+    response_2nd_line = resp.text.split("\n", 1)[1]
+    json_data = loads(response_2nd_line)["ischj"]
 
+    for item in json_data["metadata"]:
         results.append(
             {
-                'url': url,
-                'title': img_alt,
-                'content': pub_descr,
-                'source': pub_source,
-                'img_src': src_url,
-                'thumbnail_src': thumbnail_src,
+                'url': item["result"]["referrer_url"],
+                'title': item["result"]["page_title"],
+                'content': item["text_in_grid"]["snippet"],
+                'source': item["result"]["image_source_url"],
+                'img_src': item["original_image"]["url"],
+                'thumbnail_src': item["thumbnail"]["url"],
                 'template': 'images.html',
             }
         )
```

Comment:
```patch
diff --git a/searx/engines/google_images.py b/searx/engines/google_images.py
index 6d2d2e410..bca083181 100644
--- a/searx/engines/google_images.py
+++ b/searx/engines/google_images.py
@@ -104,7 +104,8 @@ def response(resp):
                 'url': item["result"]["referrer_url"],
                 'title': item["result"]["page_title"],
                 'content': item["text_in_grid"]["snippet"],
-                'source': item["result"]["image_source_url"],
+                'source': item["result"]["site_title"],
+                'format': f'{item["original_image"]["width"]} x item["original_image"]["height"]',
                 'img_src': item["original_image"]["url"],
                 'thumbnail_src': item["thumbnail"]["url"],
                 'template': 'images.html',
```

To get:
* the format
* Source: the title of the website instead of hostname.